src/game/client/draw/w3dmodeldraw.cpp

```diff
@@ -13,3 +13,1161 @@
+                // the assert
+#if 0
+                captainslog_dbgassert(!m_modelName.Is_Not_Empty() || !m_weaponBarrelInfoVec[i].empty(),
+                    "*** ASSET ERROR: No fx bone named '%s' found in model %s!",
+                    weaponfirefxbonename.Str(),
+                    m_modelName.Str());
+#endif
+            }
+        }
+
+        m_validStuff |= WEAPON_BARREL_INFO_VALID;
+    }
+}
+
+void ModelConditionInfo::Validate_Turret_Info() const
+{
+    if ((m_validStuff & TURRET_VALID) == 0) {
+        Set_FP_Mode();
+
+        for (int i = 0; i < 2; i++) {
+            TurretInfo *info = &m_turretInfo[i];
+
+            if (Should_Validate() && !m_modelName.Is_Empty()) {
+                if (info->m_turretAngleName) {
+                    if (!Find_Pristine_Bone(info->m_turretAngleName, &info->m_turretAngleBone)) {
+                        captainslog_debug("*** ASSET ERROR: TurretBone %s not found! (%s)",
+                            g_theNameKeyGenerator->Key_To_Name(info->m_turretAngleName).Str(),
+                            m_modelName.Str());
+                        info->m_turretAngleBone = 0;
+                    }
+                } else {
+                    info->m_turretAngleBone = 0;
+                }
+
+                if (info->m_turretPitchName) {
+                    if (!Find_Pristine_Bone(info->m_turretPitchName, &info->m_turretPitchBone)) {
+                        captainslog_debug("*** ASSET ERROR: TurretBone %s not found! (%s)",
+                            g_theNameKeyGenerator->Key_To_Name(info->m_turretPitchName).Str(),
+                            m_modelName.Str());
+                        info->m_turretPitchBone = 0;
+                    }
+                } else {
+                    info->m_turretPitchBone = 0;
+                }
+            } else {
+                info->m_turretAngleBone = 0;
+                info->m_turretPitchBone = 0;
+            }
+        }
+
+        if (Should_Validate()) {
+            m_validStuff |= TURRET_VALID;
+        }
+    }
+}
+
+Matrix3D const *ModelConditionInfo::Find_Pristine_Bone(NameKeyType key, int *index) const
+{
+    captainslog_dbgassert((m_validStuff & PRISTINE_BONES_VALID) != 0, "*** ASSET ERROR: bones are not valid");
+
+    if ((m_validStuff & PRISTINE_BONES_VALID) != 0) {
+        if (key) {
+            auto i = m_boneMap.find(key);
+
+            if (i != m_boneMap.end()) {
+                if (index) {
+                    *index = i->second.index;
+                }
+
+                return &i->second.transform;
+            } else {
+                if (index) {
+                    *index = 0;
+                }
+
+                return nullptr;
+            }
+        } else {
+            if (index) {
+                *index = 0;
+            }
+
+            return nullptr;
+        }
+    } else {
+        if (index) {
+            *index = 0;
+        }
+
+        return nullptr;

```

Comment:
I have seen this non-sense in the Pseudo. We can reduce this to just one
```
        if (index) {
            *index = 0;
        }

        return nullptr;
  ```
  at the end of the function.

Reply:
Makes sense to me. Fixed.